src/commands/scanner/run.ts

```diff
@@ -192,13 +192,20 @@ export default class Run extends ScannerCommand {
 			this.ux.log(messages.getMessage('output.filtersIgnoredCustom', []));
 		}
 
-		// Be liberal with the user, but do log an info message if they choose a file extension that does not match their format.
+		// If the user explicitly specified both a format and an outfile, we need to do a bit of validation there.
 		if (this.flags.format && this.flags.outfile) {
-			const derivedFormat = this.deriveFormatFromOutfile();
-			// For validation purposes, treat junit as xml.
-			const chosenFormat = this.flags.format == 'junit' ? 'xml' : this.flags.format as string;
-			if (derivedFormat !== chosenFormat) {
-				this.ux.log(messages.getMessage('validations.outfileFormatMismatch', [this.flags.format as string, derivedFormat]));
+			const inferredOutfileFormat = this.inferFormatFromOutfile();
+			// For the purposes of this validation, we treat junit as xml.
+			const chosenFormat = this.flags.format === 'junit' ? 'xml' : this.flags.format as string;
+			// If the chosen format is TABLE, we immediately need to exit. There's no way to sensibly write the output
+			// of TABLE to a file.
+			if (chosenFormat === OUTPUT_FORMAT.TABLE) {
+				throw SfdxError.create('@salesforce/sfdx-scanner', 'run', 'validations.cannotWriteTableToFile', []);
+			}
+			// Otherwise, we want to be liberal with the user. If the chosen format doesn't match the outfile's extension,
+			// just log a message saying so.
+			if (chosenFormat !== inferredOutfileFormat) {
+				this.ux.log(messages.getMessage('validations.outfileFormatMismatch', [this.flags.format as string, inferredOutfileFormat]));
```

Comment:
Previously, if you tried to specify `--format table --outfile whatever`, it would throw the out-of-date message about not having an acceptable file extension. Now, we explicitly throw an exception indicating that `table` can't be written to a file. This makes sense, because the table format is intended to be human-readable.